bluemira/builders/pf_coil.py

```diff
@@ -120,16 +121,34 @@ def build_xz(self, shape: BluemiraWire) -> list[PhysicalComponent]:
         """
         Build the xz cross-section of the PF coil.
         """
-        wp = PhysicalComponent(self.WINDING_PACK, BluemiraFace(shape))
+        wp = PhysicalComponent(
+            self.WINDING_PACK,
+            BluemiraFace(shape),
+            material=get_cached_material(
+                self.build_config.get("material", {}).get(self.WINDING_PACK)
+            ),
+        )
         idx = CoilType(self.params.ctype.value).value - 1
         apply_component_display_options(wp, color=BLUE_PALETTE["PF"][idx])
 
         ins_shape = offset_wire(shape, self.params.tk_insulation.value)
-        ins = PhysicalComponent(self.GROUND_INSULATION, BluemiraFace([ins_shape, shape]))
+        ins = PhysicalComponent(
+            self.GROUND_INSULATION,
+            BluemiraFace([ins_shape, shape]),
+            material=get_cached_material(
+                self.build_config.get("material", {}).get(self.GROUND_INSULATION)
+            ),
+        )
         apply_component_display_options(ins, color=BLUE_PALETTE["PF"][3])
 
         cas_shape = offset_wire(ins_shape, self.params.tk_casing.value)
-        casing = PhysicalComponent(self.CASING, BluemiraFace([cas_shape, ins_shape]))
+        casing = PhysicalComponent(
+            self.CASING,
+            BluemiraFace([cas_shape, ins_shape]),
+            material=get_cached_material(
+                self.build_config.get("material", {}).get(self.CASING)
```

Comment:
maybe also we could have a wrapper function that does some of this so the user only passes in a dictionary and a key, possibly harder to work out whats going on:
```python
get_cached_material_for_component(self.build_config, self.CASING)
```